internal/pkg/config/matcher_test.go

```diff
@@ -296,6 +296,38 @@ Field: qux`,
 	}
 }
 
+func TestFieldTagMatcherUnmarshalling(t *testing.T) {
+	testCases := []struct {
+		desc, yaml        string
+		shouldErrorOnLoad bool
+	}{
+		{
+			desc: "unknown field is not allowed",
+			yaml: `
+Key: good
+UnknownField: bad`,
+			shouldErrorOnLoad: true,
+		}, {
+			desc: "valid field tag config",
+			yaml: `
+Key: good
+Value: AlsoGood`,
+			shouldErrorOnLoad: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			ftm := fieldTagMatcher{}
+			err := yaml.UnmarshalStrict([]byte(tc.yaml), &ftm)
+
+			if (err != nil) != tc.shouldErrorOnLoad {
+				t.Errorf("error expectation = %v, but got err=%v", tc.shouldErrorOnLoad, err)
```

Comment:
[Got before want](https://github.com/golang/go/wiki/TestComments#got-before-want)

Reply:
Nit: why does `err=%v` not have whitespace, but `expectation = %v` does?

Reply:
Haha somebody before me wrote got after want. I'll try to correct all of them.